pkg/endpoint/endpoint.go

```diff
@@ -1642,7 +1642,11 @@ func (e *Endpoint) RunMetadataResolver(resolveMetadata MetadataResolverCB) {
 	done := make(chan struct{})
 	controllerName := fmt.Sprintf("resolve-labels-%s", e.GetK8sNamespaceAndPodName())
 	go func() {
-		<-done
+		select {
+		case <-done:
+		case <-e.aliveCtx.Done():
+			return
```

Comment:
I suspect that another option would be to add a `StopFunc` to the controller below to close `done`, though that said I don't think there should be a functional difference so I'm happy with this solution.

Reply:
Just for posterity, we could add a `StopFunc` to also close the `done` channel, but wouldn't work for the way we've set up this controller. Essentially, the configured run interval at 30s seems like it's there to provide a buffer for the initial run. Once it runs and succeeds, the controller closes `done` and returns, _and_ this would normally run again after 30s, but the goroutine is waiting for the `done` channel to be closed and removes the controller, thereby stopping it.

So having the close on `done` be inside `StopFunc` wouldn't work because we want to stop this controller as soon as it succeeds. 

I've prepared a commit to remove the interval because it is actually unnecessary. Waiting for CI to finish its run.